NitroxClient/GameLogic/Spawning/Metadata/IncubatorMetadataProcessor.cs

```diff
@@ -0,0 +1,22 @@
+﻿using NitroxModel.DataStructures.GameLogic.Entities.Metadata;
+using NitroxModel.Helper;
+using UnityEngine;
+
+namespace NitroxClient.GameLogic.Spawning.Metadata
+{
+    public class IncubatorMetadataProcessor : GenericEntityMetadataProcessor<IncubatorMetadata>
+    {
+        public override void ProcessMetadata(GameObject gameObject, IncubatorMetadata metadata)
+        {
+            Incubator incubator = gameObject.GetComponent<Incubator>();
+            incubator.powered = metadata.Powered;
+            incubator.hatched = metadata.Hatched;
+
+            bool terminalActive = (bool)incubator.ReflectionGet("terminalActive");
```

Comment:
```suggestion
            var terminalActive = incubator.ReflectionGet<bool>("terminalActive");
```

Reply:
Unless I am mistaken I do not believe this works due to how the ReflectionHelper returns the object type?

Reply:
My apologies, retracting this one and the other; I misinterpreted the intent of T in the `ReflectionHelper`s.

```
public static object ReflectionGet<T>(this T o, string fieldName, bool isPublic = false, bool isStatic = false)
//                  ^ expected T
```

Reply:
Would be cool if we could do this though, I would have thought even with T though would it not have issues with say in this case passing in bool but the class being passed in as Incubator would that have an issue with the return type?